sql/wsrep_xid.cc

```diff
@@ -179,6 +190,13 @@ wsrep::gtid wsrep_get_SE_checkpoint()
     return wsrep::gtid();
   }
 
+  if (xid.data[WSREP_XID_VERSION_OFFSET] == WSREP_XID_VERSION_3)
```

Comment:
Doing global variable initialization here does not look right. I would define a new class/struct which would contain all the fields which are stored in the XID and make `wsrep_get_SE_checkpoint()` return the struct and change `wsrep_set_SE_checkpoint(const wsrep::gtid&)` to take the new struct as an argument. The globals initialization should be done somewhere else where it is it is clear from context that it is right thing to do, e.g. after SST.